### PR TITLE
Enh add assert series equal

### PR DESCRIPTION
**Pull request by CloseChoice**
add compare method for series. First step to solve #1167

### CODE REVIEW

**Review by ritchie46 — COMMENTED** [review]
thanks @CloseChoice. Looks good. Only found one typo.

File: py-polars/polars/testing.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    check_dtype=True,
+    check_names=True,
+    check_exact=False,
+    rtol=1.0e-5,
+    atol=1.0e-8,
+    obj="Series",
+):
+    if obj == "Series":
+        if not (isinstance(left, Series) and isinstance(right, Series)):
+            raise_assert_detail(obj, "Type mismtach", type(left), type(right))
```
[the remark this answers is not in the document]
Comment (CloseChoice) [reply py-polars/polars/testing.py]: thanks for pointing that out. Fixed the typo and added a test for it.

**Comment by ritchie46** [comment]
@CloseChoice, in `/py-polars` there is a `Makefile`, if you run `make pre-commit` from there. A lot of formatting and linting is already done for you.

File: py-polars/polars/testing.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (CloseChoice) [reply py-polars/polars/testing.py]: done

**Comment by ritchie46** [comment]
Thanks @CloseChoice 

Note to self, add to ref guide.

**Pull request merged**